src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -429,12 +372,9 @@ private void VerifyCanWriteStartBatch(bool synchronousCall)
         }
 
         /// <summary>
-        /// Starts a new batch - implementation of the actual functionality.
+        /// Starts a new batch.
         /// </summary>
-        private void WriteStartBatchImplementation()
-        {
-            this.SetState(BatchWriterState.BatchStarted);
-        }

```

Comment:
This implementation deals with state management; its implementation (which doesn't differ between multipartmixed and json) should be left in the base class and, since it's only caller is the base class, it should be kept private..